specification/vmwarevirtustream/resource-manager/Microsoft.VMwareVirtustream/preview/2016-09-01-preview/vmwarevirtustream.json

```diff
@@ -0,0 +1,1304 @@
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "name": "privateCloudName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the privateCloud."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AddIdentitySourceRequest"
+            }
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/PrivateCloudResponse"
+            }
+          },
+          "202": {
+            "description": "accepted"
+          }
+        },
+        "x-ms-examples": {
+          "PrivateClouds_AddIdentitySource": {
+            "$ref": "./examples/PrivateClouds_AddIdentitySource.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.VMwareVirtustream/privateClouds/{privateCloudName}/deleteIdentitySource": {
+      "post": {
+        "operationId": "PrivateClouds_DeleteIdentitySource",
+        "x-ms-long-running-operation": true,
+        "tags": [
+          "PrivateClouds"
+        ],
+        "summary": "deletes an identity source from the private cloud",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "name": "privateCloudName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the privateCloud."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DeleteIdentitySourceRequest"
+            }
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/PrivateCloudResponse"
+            }
+          },
+          "202": {
+            "description": "accepted"
+          }
+        },
+        "x-ms-examples": {
+          "PrivateClouds_DeleteIdentitySource": {
+            "$ref": "./examples/PrivateClouds_DeleteIdentitySource.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.VMwareVirtustream/privateClouds/{privateCloudName}/getAdminCredentials": {
```

Comment:
The standard for getting credentials is a POST action prefixed with "list" - i.e. "listAdminCredentials". This means this [functionality can be utilized during template deployments](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-resource#list) (for example if you wanted to create a private cloud, fetch credentials, and then pass the credentials to some other resource like a keyvault).